packages/babel-types/src/definitions/flow.js

```diff
@@ -1,400 +1,422 @@
   visitor: ["typeAnnotation"],
-  aliases: ["Flow"],
+  aliases: ["Flow", "FlowType"],
   fields: {
-    // todo
+    typeAnnotation: validateType("Flow"),
   },
 });
 
 defineType("NumberLiteralTypeAnnotation", {
-  aliases: ["Flow"],
+  aliases: ["Flow", "FlowType"],
   fields: {
-    // todo
+    value: validate(assertValueType("number")),
   },
 });
 
 defineType("NumberTypeAnnotation", {
-  aliases: ["Flow", "FlowBaseAnnotation"],
-  fields: {
-    // todo
-  },
+  aliases: ["Flow", "FlowType", "FlowBaseAnnotation"],
 });
 
 defineType("ObjectTypeAnnotation", {
   visitor: ["properties", "indexers", "callProperties"],
-  aliases: ["Flow"],
+  aliases: ["Flow", "FlowType"],
   fields: {
-    // todo
+    properties: validate(
+      arrayOfType(["ObjectTypeProperty", "ObjectTypeSpreadProperty"]),
+    ),
+    indexers: validateOptional(arrayOfType("ObjectTypeIndexer")),
+    callProperties: validateOptional(arrayOfType("ObjectTypeCallProperty")),
+    exact: validate(assertValueType("boolean")),
   },
 });
 
 defineType("ObjectTypeCallProperty", {
   visitor: ["value"],
   aliases: ["Flow", "UserWhitespacable"],
   fields: {
-    // todo
+    value: validateType("Flow"),
+    static: validate(assertValueType("boolean")),
   },
 });
 
 defineType("ObjectTypeIndexer", {
-  visitor: ["id", "key", "value"],
+  visitor: ["id", "key", "value", "variance"],
   aliases: ["Flow", "UserWhitespacable"],
   fields: {
-    // todo
+    id: validateOptionalType("Identifier"),
+    key: validateType("Flow"),
+    value: validateType(["Flow", "Identifier"]),
+    static: validate(assertValueType("boolean")),
+    variance: validateOptionalType("Variance"),
   },
 });
 
 defineType("ObjectTypeProperty", {
-  visitor: ["key", "value"],
+  visitor: ["key", "value", "variance"],
   aliases: ["Flow", "UserWhitespacable"],
   fields: {
-    // todo
+    key: validateType("Identifier"),
+    value: validateType(["Flow", "Identifier"]),
+    kind: validate(assertOneOf("init", "get", "set")),
+    static: validate(assertValueType("boolean")),
+    optional: validate(assertValueType("boolean")),
+    variance: validateOptionalType("Variance"),
   },
 });
 
 defineType("ObjectTypeSpreadProperty", {
   visitor: ["argument"],
   aliases: ["Flow", "UserWhitespacable"],
   fields: {
-    // todo
+    argument: validateType("Flow"),
   },
 });
 
 defineType("OpaqueType", {
   visitor: ["id", "typeParameters", "supertype", "impltype"],
   aliases: ["Flow", "FlowDeclaration", "Statement", "Declaration"],
   fields: {
-    // todo
+    id: validateType("Identifier"),
+    typeParameters: validateOptionalType("TypeParameterDeclaration"),
+    supertype: validateOptionalType("Flow"),
+    impltype: validateType("Flow"),
   },
 });
 
 defineType("QualifiedTypeIdentifier", {
   visitor: ["id", "qualification"],
   aliases: ["Flow"],
   fields: {
-    // todo
+    id: validateType("Identifier"),
+    qualifications: validateType("Identifier"),
```

Comment:
This should be `quailfication` (without the `s`). Also, it should be `Identifier | QualifiedTypeIdentifier`: https://github.com/babel/babel/blob/ad2019aa303fd0de6145f0cd5ddf264ab831d590/packages/babylon/test/fixtures/flow/declare-statements/16/expected.json#L92-L93

Reply:
Good catch!